src/zenml/utils/pydantic_utils.py

```diff
@@ -0,0 +1,174 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utilities for pydantic models."""
+import json
+from typing import Any, Dict, Optional, Type, TypeVar, Union, cast
+
+from pydantic import BaseModel
+from pydantic.json import pydantic_encoder
+from pydantic.utils import sequence_like
+
+from zenml.utils import dict_utils
+
+M = TypeVar("M", bound="BaseModel")
+
+
+def update_model(
+    original: M,
+    update: Union["BaseModel", Dict[str, Any]],
+    recursive: bool = True,
+    exclude_none: bool = True,
+) -> M:
```

Comment:
very curious to see if we can unify this with zenml.zenserver.models where we do something similar in the megabranch